contracts/RFQ.sol

```diff
@@ -51,111 +54,125 @@ contract RFQ is IRFQ, Ownable, TokenCollector, EIP712 {
     }
 
     function fillRFQ(
-        Offer calldata offer,
+        RFQOffer calldata rfqOffer,
+        uint256 takerRequestAmount,
         bytes calldata makerSignature,
         bytes calldata makerTokenPermit,
         bytes calldata takerTokenPermit,
         address payable recipient,
         uint256 feeFactor
```

Comment:
Q: Why not using `RFQTx` here? I see the relevant args are all needed to be passed in too.